pkg/operator/deploy/deploy.go

```diff
@@ -374,53 +374,51 @@ func (o *operator) IsReady(ctx context.Context) (bool, error) {
 	return true, nil
 }
 
-func checkOperatorDeploymentVersion(ctx context.Context, cli appsv1client.DeploymentInterface, name string, gitCommit string) error {
+func checkOperatorDeploymentVersion(ctx context.Context, cli appsv1client.DeploymentInterface, name string, gitCommit string) (bool, error) {
 	d, err := cli.Get(ctx, name, metav1.GetOptions{})
 	if err != nil {
-		return err
+		return false, err
 	}
-
 	if d.Labels["version"] != gitCommit {
-		return errors.New(name + " is not running the desired version: " + gitCommit)
+		return false, errors.New(name + " is not running the desired version: " + gitCommit)
 	}
-
-	return nil
+	return true, nil
 }
 
-func checkPodImageVersion(ctx context.Context, cli corev1client.PodInterface, namespace string, gitCommit string) error {
+func checkPodImageVersion(ctx context.Context, cli corev1client.PodInterface, namespace string, gitCommit string) (bool, error) {
 	podList, err := cli.List(ctx, metav1.ListOptions{LabelSelector: "app=" + namespace})
 	if err != nil {
-		return err
+		return false, err
 	}
```

Comment:
Again, the pod may not yet be running.  If it doesn't exist, this will error out.  If it doesn't exist, just return false, nil

```suggestion
	switch {
		case kerrors.IsNotFound(err):
			return false, nil
		case err != nil:
			return false, err
		}